### PR TITLE
readme: bump up golang version

### PR DESCRIPTION
**Pull request by nonsense**
Updating README to ask for Go version 1.20.x instead of 1.19.x

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,10 +71,10 @@ For other distributions you can find the required dependencies [here.](https://l
 
 #### Go
 
-To build Lotus, you need a working installation of [Go 1.19.12 or higher](https://golang.org/dl/):
+To build Lotus, you need a working installation of [Go 1.20.10 or higher](https://golang.org/dl/):
```
Comment (fridrik01) [inline README.md:74]: Lotus supports Go 1.19.12 so this is not necessarily correct

Comment (Stebalien) [reply README.md:74]: @nonsense bump

Comment (nonsense) [reply README.md:74]:
```
$ go version
go version go1.19.2 darwin/arm64
```

```
~/code/src/github.com/filecoin-project/lotus (main)
$ make clean all
Makefile:14: Your Golang version is go1.19.2
Makefile:15: *** Update Golang to version to at least 1.20.7.  Stop.
```

See: https://github.com/filecoin-project/lotus/blob/master/GO_VERSION_MIN

Comment (Stebalien) [reply README.md:74]: Ah, you're right. I was looking at the release branch.

**Review by Stebalien — APPROVED** [review]

**Pull request merged**